runtime/ts/storage/in-memory-storage.ts

```diff
@@ -155,22 +148,39 @@ abstract class InMemoryStorageProvider extends StorageProviderBase {
     }
     return new InMemoryVariable(type, storageEngine, name, id, key);
   }
+
+  async ensureBackingStore() {
```

Comment:
It seems most call sites await (or .then) the return from this - do we need to have the internal awaits as well?

Reply:
There's one asynchronous requirement - this.backingStore needs to await the output of this.storageEngine.baseStorageFor. However, that can happen in a `then()`, which means we can directly return this.pendingBackingStore. Awaiting this function guarantees that this.backingStore will exist afterwards, as the `then()` assignment must execute before anything scheduled after that await.